Add improved Pollux system

syonfox opened the pull request:
<!-- Please describe new feature, possible with screenshot if needed. -->
<!-- Please make sure you've read documentation on contributing -->

* New faction Syon Free Traders & Company 
* New Home worlds in sector 4, 2, 5 Called Syon570 
* Updated Pollux system to include its large gas giant and 2 made up but plausible moons with a refiling station between them.

https://web.archive.org/web/20071106060030/http://exoplanet.eu/star.php?st=HD+62509 
https://en.wikipedia.org/wiki/Pollux_(star)

![image](https://github.com/user-attachments/assets/8e8b260d-ff1f-4a50-b7a7-c093fe74927a)

2 bugs I see is Hd265866 is a duplicate of Gliese 251 https://en.wikipedia.org/wiki/Gliese_251

and Alpha Geminorum is a duplicate of Castor.  Ill try and add these to the list I also want to add https://en.wikipedia.org/wiki/Rho_Geminorum as it completes https://en.wiktionary.org/wiki/%E5%8C%97%E6%B2%B3%E4%B8%89 

But now Im getting ahead of myself.  These systems will need some more work to get right as I am learning more about astronomy and how to decipher the system descriptions.

sturnclaw reviewed (changes requested):
I appreciate the effort in improving the Pollux system! However, I'm going to ask you to separate the Pollux changes from the addition of the 'Syon' faction and system if you're interested in continuing with the PR, for reasons outlined below:

Firstly, the addition of a new faction with associated homeworld is quite orthogonal to the stated goal of improving the Pollux system - the Pioneer project as a whole has a goal of trying to reduce the number of factions and make the factions that exist more fleshed-out, rather than adding new factions.

Additionally, the proposed faction name and lore are something I am not willing to merge due to conflicts with the established lore conventions of Pioneer.

To my knowledge, this would be the *only* faction in the game named after a contributor rather than an in-universe polity, and consequently the proposed faction is quite the "4th wall"-break - and that's without getting into the ramifications of a tiny interstellar polity controlling an ostensibly-neutral system in what is the in-universe equivalent of the Korean DMZ.

Though I understand it may be unintentional on your part, asking to "canonize" a faction and star system directly named after (multiple parts of!) your own name/handle can come across as somewhat rude.

Historically, we've reserved permanently naming even things as minor as space stations to only contributors with a very significant history with the project - Tomm's Sanctuary is named after the original sole author of the game before it was open-sourced, for example. Otherwise, contributor names are only ever used as "easter eggs" through the randomly-generated name pool.

Please feel free to keep the Syon faction and system as a mod - I'd love to see the mod tooling be used and the bugs in it found! - but for the reasons above it's not something I'm willing to sign off on directly adding to the official Pioneer universe.

If you're interested in improving or adding to Pioneer's lore (i.e. writing replacements for the existing procedurally-generated factions, adding new systems, etc.), I'd encourage you to propose changes and ask for feedback on IRC so that your efforts can align with the intended creative direction for Pioneer - we'd love to have contributors interested in working on that aspect of the game!

syonfox commented:
Fair enough yeah I was not sure what your thoughts on the faction were I was trying to fit it into the current lore in a similar vein to the Harbor corperation or other except for some mix of sentient ai and pirate coalition.  But I guess it is a little forward for the unbiased game.

syonfox commented:
Ill plan to continue the syon faction as a mod you can find it over at https://github.com/syonfox/eapi-space

As of writing it is a ways away from the first release but is working with the current game (it runs ;)

syonfox commented:
Ok a little back on topic for the Gliese 521 system I noticed the position is different in each database as parsed by the last guy.  And I noticed the distance claimed in wikipidia is not the same as the one in pioneer. 

So the question is . Given galactic coordinates and distance. 

How can we compute the sector and position? 

Do we have internal code for this? Is there an online tool? what is the correct math?


### ie:

Gliese 251

[Right ascension](https://en.wikipedia.org/wiki/Right_ascension) 	06h 54m 48.95806s[[1]](https://en.wikipedia.org/wiki/Gliese_251#cite_note-GaiaDR3-1)
[Declination](https://en.wikipedia.org/wiki/Declination) 	+33° 16′ 05.4383″[[1]](https://en.wikipedia.org/wiki/Gliese_251#cite_note-GaiaDR3-1)
[Distance](https://en.wikipedia.org/wiki/Distance_(astronomy))	18.215 ± 0.003 [ly](https://en.wikipedia.org/wiki/Light-year)


```
function galactic_pos_to_pioneer(right_asention, declanation, distance) {
  //??
  return {sector, pos}
}
```

sturnclaw commented:
I do not believe we have a set of algorithms yet to convert between GCS and Pioneer's coordinate system. It's on my TODO list to implement for #4879, but definitely will not make it for this release.

syonfox commented:
So in stellarium you can copy object info like `ctrl+shift+c`
```
HIP 33226 - HD 265866
Type: star
Magnitude: 9.85
Absolute Magnitude: 11.12
Color Index (B-V): 1.58
RA/Dec (J2000.0): 6h54m47.48s/+33°15'56.1"
RA/Dec (on date): 6h56m25.99s/+33°14'04.0"
HA/Dec: 14h41m16.25s/+33°14'04.0" 
Az./Alt.: +32°47'58.7"/-2°28'58.5" 
Gal. long./lat.: -177°03'47.7"/+15°07'46.8"
Supergal. long./lat.: +26°23'16.7"/-41°01'31.6"
Ecl. long./lat. (J2000.0): +101°36'44.8"/+10°22'36.6"
Ecl. long./lat. (on date): +101°57'46.2"/+10°22'47.9"
Ecliptic obliquity (on date): +23°26'18.6"
Mean Sidereal Time: 21h37m42.2s
Apparent Sidereal Time: 21h37m42.2s
Rise: 14h42m
Transit: 23h37m
Set: 8h32m
IAU Constellation: Gem
Distance: 18.20±0.01 ly
Proper motion: 830.8 mas/yr towards 241.3°
Proper motions by axes: -728.5 -399.3 (mas/yr)
Parallax: 179.160±0.063 mas
Spectral Type: M3V
```

Ill do some diggin and see

But it might be nice to get stars from stellarium as its a massive project with lots of people depending on it for accuracy of telescope positioning etc. 


... I was able get some approx calculations working as far as my test show there is just a +-2 degree declination error fro the existing systems.  It seems that pioneer is using 

j2000 Ra/Dec but with an offset of -90 degrees to Ra

some results: 

```bash
Equatorial Coordinates From Stellarium (RA, Dec): { ra: 103.69783333333334, dec: 33.26558333333333 }
Actual Pos (x,y,z):  [ 15.256, 3.72, 10.312 ]
{ x: 15.256, y: 3.72, z: 10.312 }
Pioneer Pos Equ:  { ra: 13.70350265941236, dec: 34.055971253129265 }
Stellar Pos Equ:  { ra: 103.69783333333334, dec: 33.26558333333333 }
[End test debug]
stellar_input_w_offset_ra (ra,dec,dist) : 13.697833333333335 33.26558333333333 18.2
stellar_pos_calculated:  { x: 15.217693397544485, y: 3.6035711973706452, z: 9.983076061932563 }
approx_pos_form_pioneer:  { x: 15.256, y: 3.72, z: 10.312 }
stellar_pioneer_output:  {
  sector: [ 1, 0, 1 ],
  pos: [ 7.217693397544485, 3.603571197370645, 1.9830760619325627 ]
}
original_pioneer_input:  { sector: [ 1, 0, 1 ], pos: [ 7.256, 3.72, 2.312 ] }
Actual Pos (x,y,z):  [ 8.08, 1.608, -2.4720000000000004 ]
{ x: 8.08, y: 1.608, z: -2.4720000000000004 }
Pioneer Pos Equ:  { ra: 11.255375473261594, dec: -17.01099836882489 }
Stellar Pos Equ:  { ra: 101.28270833333333, dec: -15.275 }
[End test debug]
stellar_input_w_offset_ra (ra,dec,dist) : 11.282708333333332 -15.275 8.6
stellar_pos_calculated:  { x: 8.296183181570512, y: 1.6231498094069947, z: -2.2656885526979544 }
approx_pos_form_pioneer:  { x: 8.08, y: 1.608, z: -2.4720000000000004 }
stellar_pioneer_output:  {
  sector: [ 1, 0, -1 ],
  pos: [ 0.2961831815705125, 1.6231498094069945, 5.734311447302046 ]
}
original_pioneer_input:  { sector: [ 1, 0, -1 ], pos: [ 0.08, 1.608, 5.528 ] }
[
  [
    'gliese 251',
    {
      deltaDec: 0.7903879197959327,
      deltaRa: -89.99433067392097,
      ra: 13.697833333333335,
      dec: 33.26558333333333,
      pos: [Object]
    }
  ],
  [
    'sirius',
    {
      deltaDec: -1.7359983688248892,
      deltaRa: -90.02733286007174,
      ra: 11.282708333333332,
      dec: -15.275,
      pos: [Object]
    }
  ]
]

```

sturnclaw commented on data/systems/custom/24_gliese_251.json:
```diff
@@ -0,0 +1,497 @@
+	"sector": [ 1, 0, 1 ],
+	"pos": [ 7.217693397544485, 3.603571197370645, 1.9830760619325627 ],
+	"seed": 3464555875,
+	"shortDesc": "Located in the constellation of Gemini, it is the nearest star ro Sol in this constellation.",
```
```suggestion
	"shortDesc": "Located in the constellation of Gemini, it is the nearest star to Sol in this constellation.",
```

sturnclaw commented on data/systems/custom/24_gliese_251.json:
```diff
@@ -0,0 +1,497 @@
+	"explored": true,
+	"govType": "NONE",
+	"lawlessness": "f0/4284095261",
+	"longDesc": "Gliese 251 is a red dwarf with a spectral type of M3V[3] with an effective temperature of about 3300 K.[3] Its mass has been measured to be around 0.36 solar masses[6] and its radius is about 36% solar radii.[6] Its metallicity is likely slightly less than that of the Sun.[7] Observations at infrared wavelengths rule out the presence of a circumstellar disk around it.",
```
Please remove the footnote annotations from this description.

syonfox replied:
Fixed

sturnclaw reviewed (changes requested):
You'll need to remove the `103_Syon_Traders_And_Company.lua` file from the PR.

I'll look over the changes to Gliese 251 and Pollux tomorrow.

syonfox commented:
Okay, updated thanks for the review.

sturnclaw commented on data/systems/custom/22_pollux.json:
```diff
@@ -0,0 +1,263 @@
+	"explored": true,
+	"govType": "NONE",
+	"lawlessness": "f0/403838384",
+	"longDesc": "Pollux, the brighter star of the Gemini constellation, hosts a simple and stable system in contrast to its twin, Castor, a chaotic dance of six stars. Around Pollux orbits the gas giant Thestias, accompanied by two moons. Between these moons, Foxi Co. built Stelida Station, a safe haven for travelers among the stars.\n\nWhen Castor died, Pollux was devastated. Zeus offered to reunit ...
```
```suggestion
	"longDesc": "Pollux, the brighter star of the Gemini constellation, hosts a simple and stable system in contrast to its twin, Castor, a chaotic dance of six stars. Around Pollux orbits the gas giant Thestias, accompanied by two moons. Between these moons sits Stelida Station, a safe haven for travelers among the stars.\n\nWhen Castor died, Pollux was devastated. Zeus offered to reunite them by granting Castor partial immortality. Together, they were placed in the heavens as the constellation Gemini, symbolizing their eternal brotherhood.\n\nIn Chinese astronomy, Pollux is known as 北河三 (Běi Hé sān), meaning \"Third Star of the North River.\" It belongs to the asterism 北河 (Běi Hé), or \"North River,\" alongside Castor and Rho Geminorum.",
```

sturnclaw reviewed (changes requested):
There is an errant reference to the removed custom faction remaining in the description for Pollux.

impaktor commented:
Lest I forget, if/when this is merged, do you want to be in [AUTHORS.txt](https://github.com/pioneerspacesim/pioneer/blob/master/AUTHORS.txt), and by what name?

syonfox commented:
@impaktor sure it cant hurt I suppose.  I'd prefer syonfox

syonfox commented on data/systems/partial/02_local_stars.json:
```diff
@@ -16,7 +16,6 @@
     {"name": "Gliese 486", "stars": ["STAR_M"], "sector": [-1, 3, 0], "pos": [2.52, 1.816, 4.536]},
     {"name": "NN 3417", "stars": ["STAR_M"], "sector": [1, 0, 2], "pos": [3.368, 2.936, 6.392]},
     {"name": "YZ Ceti", "stars": ["STAR_M"], "sector": [0, -2, -1], "pos": [3.608, 4.984, 4.456]},
-    {"name": "Gliese 251", "stars": ["STAR_M"], "sector": [1, 0, 1], "pos": [7.256, 3.72, 2.312]},
```
Remove entry 1

syonfox replied:
Will try returning this and see if game works ... it does

syonfox commented on data/systems/partial/02_local_stars.json:
```diff
@@ -316,7 +315,6 @@
     {"name": "Gliese 694", "stars": ["STAR_M"], "sector": [-3, 0, 2], "pos": [1.56, 1.568, 5.256]},
     {"name": "Gliese 275.2", "stars": ["STAR_M", "WHITE_DWARF"], "sector": [2, 1, 3], "pos": [7.176, 1.68, 4.12]},
     {"name": "NN 3233", "stars": ["STAR_G"], "sector": [3, -3, 3], "pos": [7.584, 0.272, 6.936]},
-    {"name": "HD 265866", "stars": ["STAR_K"], "sector": [1, 0, 1], "pos": [6.608, 3.552, 1.864]},
```
Removed entry 2

syonfox commented on data/systems/custom/24_gliese_251.json:
Added custom system

syonfox commented on data/systems/partial/02_local_stars.json:
Pollux is still in here at line 85 but in my star map this is not breaking anything. 
Does this mean the Custom system is now overriding systems with the same name?

syonfox replied:
Ok will do, but as a note it seems to be working just tested.

syonfox reviewed (commented):
Ok looking good just testing in game and it looks like it is working properly the star map doesn't show duplicate even thou the entry is still in local stars. I think this is the desired behavior so i will leave it for testing.

syonfox commented:
Just did a play test and they they are fun and realistic systems. Seems to be working well. LMK if the squash worked 

tldr:

- Added custom Pollux system with some nice lore from astrology 
- Replaced duplicate Gliese 251 with custom system with a single asteroid observation post on the super earth. 
- Removed the entries from local_stars.json

zonkmachine commented on data/systems/custom/24_gliese_251.json:
```diff
@@ -0,0 +1,497 @@
+	"explored": true,
+	"govType": "NONE",
+	"lawlessness": "f0/4284095261",
+	"longDesc": "Gliese 251 is a red dwarf with, It is located near the boundary with Auriga, 49 arcminutes away from the bright star Theta Geminorum; due to its apparent magnitude of +9.89 it cannot be observed with the naked eye from Sol.  Its metallicity is likely slightly less than that of the Sun. Observations at infrared wavelengths rule out the presence of a circumstellar disk around it.",
```
` a red dwarf with, It is located `

With what? It looks like some text got lost here.

`49 arcminutes away`

As seen from Earth in telescopes. But people go there in ships. How far in light years?

`Its metallicity is likely slightly less than that of the Sun.`

We have access to a whole observation post luckily, so we have good measurements of the metallicity today in the year 3225. Just remove 'likely' from this sentence.  ;)

`Observations at infrared wavelengths rule out the presence of a circumstellar disk around it.`

I'm in the system right now. There is no disc of any form here. Please remove this comment.

syonfox replied:
Ok yeah, but where are the people playing the game from?  I understand the goal is to be unbiased. ie "the pioneer universe does not care about the player." 

So I guess what your saying is any sol centric descriptions should be limited to sol local region. With this the stars that are seen from sol may have inspired mission. 

So yeah as I am a monkey from sol its important to check the bias of this importance for the game. as the angles of stars are how we measure them to put them into the game it is not so important in the actual game.

A nice way of saying this might be. Even if the stars align from sol to the rest of the universe they are unaligned.

I think this will be important to remember as the stars in the game that are defined should be defined in an unbiased way as close to the truth that we know.

That said if its all unbiased it is all uniform and may become boring. 

So how do we find the balance of spice and lore. Ill maybe make an issue to discuse this more in some time.


What do you think of?

> After the reduction of the Solar Federations expansion θgem became a safe haven for merchants and traders. A moto is written on the outpost wall "Steloj nevicigitaj (Δθ!=0) estas perfekte en loko."

"stars unaligned(Δθ!=0) are perfectly in place." in esparanto

long version: 

> In the initial expansion of the Solar Federation a city was created on a moon of the main planet of Theta Gem. Since then the ability to project power this far out has deteriorated. Due to a strong economic output of the city and manufacturing base the city became independent relying on informal agreements with the nearby systems it maintains its independence since then although its reach only extends to the city limits it remains a small gem among the stars. You will find all sorts at the distant shanty outpost but ensure you bring fuel and fly safe.

zonkmachine commented on data/systems/custom/23_thetagem.json:
```diff
@@ -0,0 +1,416 @@
+			"volcanicity": "f0/13128512"
+		}
+	],
+	"comment": "http://stars.astro.illinois.edu/sow/thetagem.html",
```
What license is this text under?

impaktor commented on data/systems/custom/23_thetagem.json:
```diff
@@ -0,0 +1,416 @@
+		13
+	],
+	"seed": 34,
+	"shortDesc": "Theta Geminorum, powered by a single Type A sub-giant star. Despite its distance it shines at 101 times the brightness of Sol.",
```
Despite its distance to what?

syonfox replied:
Yeah I guess this is outdated. will push a playable tested version in a bit with some more pioneer themed lore that is less sol centric.
